pkg/api/register.go

```diff
@@ -78,12 +78,12 @@ func init() {
 
 	// Register Unversioned types
 	// TODO this should not be done here
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.ListOptions{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.Status{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.APIVersions{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.APIGroupList{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.APIGroup{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.APIResourceList{})
+	Scheme.AddKnownTypes(unversioned.SchemeGroupVersion, &unversioned.ListOptions{})
```

Comment:
IMO, all registration in pkg/api/register.go should be registered to the api.ShemeGroupVersion, so I think we should keep the original code.

We can move the registrations to `unversioned.SchemeGroupVersion` to the pkg/api/unversioned/register.go, that will be logical,  though that just duplicates the registration.